metatron-app/metatron-discovery
# F#2659 add metadata personal favorite feature

### Description

Add personal favorite feature
- catalog favorite
- metadata favorite

**Related Issue** : 

#2659 

### How Has This Been Tested?


1. Go to Exploration data 
2. check catalog favorite list is showing properly in list mode
3. check when click catalog favorite icon, favorite icon color is toggled and favorite catalog list in lnb is changing properly in list mode
4. check metadata favorite icon toggle properly in list mode
5. check favorite icon in metadata detail modal works properly
6. check my favorite metadata list in main shows data properly
#### Need additional checks?


### Types of changes



### Checklist:


### Additional Context
